mlir/utils/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,357 @@
+    }
+    stages {
+        stage("Builds and tests") {
+            parallel {
+                stage("Shared library") {
+                    agent {
+                        docker {
+                            image dockerImage()
+                            args dockerArgs()
+                            label "${params.canXdlops ? 'gfx908' : 'rocm' } && multi_gpu"
+                            alwaysPull true
+                        }
+                    }
+                    environment {
+                        PATH="$PATH:/opt/rocm/llvm/bin"
+                    }
+                    stages {
+                        stage('Environment') {
+                            steps {
+                                showEnv()
+                            }
+                        }
+                        stage("Shared library build and test") {
+                            steps {
+                                buildProject('check-mlir', """
+                               -DMLIR_MIOPEN_DRIVER_PR_E2E_TEST_ENABLED=${params.nightly ? '0' : '1'}
+                               -DMLIR_MIOPEN_DRIVER_XDLOPS_TEST_ENABLED=${params.canXdlops ? '1' : '0'}
+                               -DMLIR_MIOPEN_DRIVER_E2E_TEST_ENABLED=${params.nightly ? '1' : '0'}
+                               -DMLIR_MIOPEN_DRIVER_MISC_E2E_TEST_ENABLED=${params.nightly ? '1' : '0'}
+                               -DCMAKE_EXPORT_COMPILE_COMMANDS=1
+                               """)
+                            }
+                        }
+                        stage('Static Tests') {
+                            when { allOf {
+                                equals expected: false, actual: params.nightly;
+                                // Avoid this test running twice
+                                equals expected: false, actual: params.canXdlops
+                            } }
+                            steps {
+                                sh '''
+                                if [ ! -f ./build/compile_commands.json ];  then
+                                  echo "No compile commands, bailing."
+                                  exit 1
+                                fi
+                                if [ ! -f ./compile_commands.json ]; then
+                                  ln -s build/compile_commands.json compile_commands.json
+                                fi
+                                '''
+                                sh 'python3 ./mlir/utils/jenkins/static-checks/premerge-checks.py'
+                            }
+                        }
+                        stage("Random E2E test") {
+                            when {
+                                equals expected: true, actual: params.nightly
+                            }
+                            steps {
+                                sh 'rm -rf build' // Permissions issues when using deleteDir()
```

Comment:
This would delete all -D variables. For Random E2E tests, we still need  
 -DMLIR_MIOPEN_DRIVER_E2E_TEST_ENABLED=1 and 
 -DMLIR_MIOPEN_DRIVER_XDLOPS_TEST_ENABLED=1/0.

Reply:
Good point, I'll get rid of it